src/main/java/ai/shreds/infrastructure/InfrastructureSupplierRepositoryImpl.java

```diff
@@ -0,0 +1,87 @@
+
+    @Autowired
+    private EntityManager entityManager;
+
+    /**
+     * Finds all suppliers based on the provided filtering criteria.
+     * 
+     * @param params The filtering criteria for retrieving suppliers.
+     * @return A list of suppliers that match the filtering criteria.
+     */
+    @Override
+    public List<DomainSupplierEntity> findAll(SharedRequestParams params) {
+        logger.info("Finding all suppliers with params: {}", params);
+        String queryStr = buildQueryString(params);
+        TypedQuery<DomainSupplierEntity> query = entityManager.createQuery(queryStr, DomainSupplierEntity.class);
+        if (params.getName() != null && !params.getName().isEmpty()) {
+            query.setParameter("name", "%" + params.getName() + "%");
+        }
+        if (params.getContact_info() != null && !params.getContact_info().isEmpty()) {
+            query.setParameter("contact_info", "%" + params.getContact_info() + "%");
+        }
+        if (params.getAddress() != null && !params.getAddress().isEmpty()) {
+            query.setParameter("address", "%" + params.getAddress() + "%");
+        }
+        List<DomainSupplierEntity> result = query.getResultList();
+        logger.info("Found {} suppliers", result.size());
+        return result;
+    }
+
+    /**
+     * Constructs the query string based on the provided filtering criteria.
+     * 
+     * @param params The filtering criteria for retrieving suppliers.
+     * @return The constructed query string.
+     */
+    private String buildQueryString(SharedRequestParams params) {
+        StringBuilder queryStr = new StringBuilder("SELECT s FROM DomainSupplierEntity s WHERE 1=1");
+        if (params.getName() != null && !params.getName().isEmpty()) {
+            queryStr.append(" AND s.name LIKE :name");
+        }
+        if (params.getContact_info() != null && !params.getContact_info().isEmpty()) {
+            queryStr.append(" AND s.contact_info LIKE :contact_info");
+        }
+        if (params.getAddress() != null && !params.getAddress().isEmpty()) {
+            queryStr.append(" AND s.address LIKE :address");
+        }
+        return queryStr.toString();
+    }
+
+    /**
+     * Finds a supplier by its unique ID.
+     * 
+     * @param id The unique identifier of the supplier.
+     * @return The supplier entity with the specified ID.
+     * @throws SupplierNotFoundException If no supplier is found with the specified ID.
+     */
+    @Override
+    public DomainSupplierEntity findById(Long id) {
+        logger.info("Finding supplier by ID: {}", id);
+        return Optional.ofNullable(entityManager.find(DomainSupplierEntity.class, id))
+                      .orElseThrow(() -> new SupplierNotFoundException("Supplier not found."));
+    }
+
+    private static class SupplierNotFoundException extends RuntimeException {
+        public SupplierNotFoundException(String message) {
+            super(message);
+        }
+    }
```

Comment:
remove the SupplierNotFoundException declaration in the InfrastructureSupplierRepositoryImpl class and use the SupplierNotFoundException from the ai.shreds.domain package